HB/common/database.elpi

```diff
@@ -256,12 +256,16 @@ get-cs-structure (cs-instance _ _ (global Inst)) Struct :- std.do! [
   coq.prod-tgt->gref InstTy Struct
 ].
 
-pred has-cs-instance i:gref, i:cs-instance.
-has-cs-instance GTy (cs-instance _ (cs-gref GTy) _).
+pred has-cs-instance i:term, i:cs-instance.
+has-cs-instance (global GR) (cs-instance _ (cs-gref GR) _) :- !.
+has-cs-instance (app[global GR|_]) (cs-instance _ (cs-gref GR) _) :- !.
+has-cs-instance (prod _ _ _) (cs-instance _ cs-prod _) :- !.
+has-cs-instance (sort U) (cs-instance _ (cs-sort U) _) :- !.
```

Comment:
```suggestion
has-cs-instance (sort U) (cs-instance _ (cs-sort U) _) :- !.
has-cs-instance (fun _ _ _) (cs-instance _ _ /*cs-fun*/ _) :- coq.version _ _ N _, N < 14, coq.error "Coq cannot handle canonical values which are anonymous functions".
has-cs-instance (fun _ _ _) (cs-instance _ _ /*cs-fun*/ _) :- coq.version _ _ N _, N >= 14, coq.error "NYI: canonical values which are anonymous functions".
```

Reply:
@CohenCyril is this what you want? It is unclear what to do when we move to Coq 8.14 (assuming it gets support for `cs-fun` canonical value)